jvm/sargon-android/src/main/java/com/radixdlt/sargon/os/driver/AndroidBiometricAuthorizationDriver.kt

```diff
@@ -87,24 +48,24 @@ class BiometricsHandler(
     private val biometricsResultsChannel = Channel<Result<Unit>>()
 
     fun register(activity: FragmentActivity) {
-        activity.lifecycleScope.launch {
-            // Listen to biometric prompt requests while the activity is at least started.
-            activity.lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
-                biometricRequestsChannel.receiveAsFlow().collectLatest {
-                    val result = requestBiometricsAuthorization(activity)
-
-                    // Send back the result to sargon os
-                    biometricsResultsChannel.send(result)
-                }
+        biometricRequestsChannel
+            .receiveAsFlow()
+            .flowWithLifecycle(
+                lifecycle = activity.lifecycle,
+                minActiveState = Lifecycle.State.STARTED
+            )
+            .onEach {
+                val result = requestBiometricsAuthorization(activity)
+
+                // Send back the result to sargon os
+                biometricsResultsChannel.send(result)
```

Comment:
sure, but the comment then is wrong? this result does not reach the Rust SargonOS, no?

Reply:
It does, in order for the exception to reach the host, it needs to pass through sargon os. Remember this code is executed when sargon requests to read/write something through its secure driver. In order for the host to learn about the error it needs to pass through sargon rust.

Reply:
I don't mean to be pedantic about this, but there is no such driver as BiometricAuth as far SargonOS is concerned. This result is sent to SecureStorage driver, after that it only is passed to SargonOS, no?

Reply:
It is yes, passed to sargon os, but then a function which uses the driver passes the result back to the host.

For example the function

`sargonOs.newWallet()` can throw such exception which will have travelled all the way from the biometrics handler to the driver to ffi, to sargon rust and back to host's ui

Reply:
I understand this, anyway, I am not stubborn about this, just that when reading it for the first time I thought that the result is sent to SargonOS directly and I was confused since there is no such driver in Rust OS.

Reply:
I know the API is difficult to read, since what we are trying to do here is to ask for reading/writing something from sargon rust to the host, without asking for biometrics prior to it. So sargon os is oblivious to asking for biometrics, and does not count for android that does not automatically request for biometrics.

In order to ask for biometrics, it requires communication with the UI layer, and in order to do that with a suspend manner, we need to involve 2 different channels. This result channel sends the result of the biometrics back to the channel which, the secure driver is "subscribed" to and waiting for the response in order to either encrypt/decrypt or fail and communicate to the os why it failed. A simple `FailedToWriteMnemonic` error does not suffice since the host needs to display why it got the error to the user. 

In the case were the error is that the driver failed to encrypt, because the user just cancelled the biometrics, does not need to display that error to the user. But if the error is due to the fact that the user has not enrolled to biometrics, we need to display that to the user, instead of a generic error "Could not create a new wallet" or something similar.

The reason it needs to communicate this failure to the os, is because the host also needs to learn about it.